src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc

```diff
@@ -0,0 +1,687 @@
+        auto [replacement_l, replacement_r, replacement_pivot] = tuple;
+        if (!extent_will_overflow(replacement_pivot.size() + 1, std::nullopt)) {
+          journal_inner_update(liter, replacement_l->get_laddr(), maybe_get_delta_buffer());
+          journal_inner_replace(riter, replacement_r->get_laddr(),
+                                replacement_pivot, maybe_get_delta_buffer());
+          std::list<laddr_t> dec_laddrs;
+          dec_laddrs.push_back(l->get_laddr());
+          dec_laddrs.push_back(r->get_laddr());
+          return oc.tm.dec_ref(oc.t, dec_laddrs).safe_then([] (auto &&ret) {
+            return merge_entry_ret(
+                   merge_entry_ertr::ready_future_marker{},
+                   mutation_result_t(mutation_status_t::SUCCESS, std::nullopt, std::nullopt));
+          });
+        } else {
+          logger().debug("{}::merge_entry balanced and split {} r {}", __func__, *l, *r);
+          return make_split_children(oc).safe_then([=] (auto p_tuple) {
+            OMapNodeRef pleft = std::get<0>(p_tuple);
+            OMapNodeRef pright = std::get<1>(p_tuple);
+            std::string p_pivot = std::get<2>(p_tuple);
+            OMapInnerNodeRef p_left = pleft->cast<OMapInnerNode>();
+            OMapInnerNodeRef p_right = pright->cast<OMapInnerNode>();
+            auto [replacement_l, replacement_r, replacement_pivot] = tuple;
+            if (riter.get_index() < p_left->get_node_size()) {
+              auto mut_liter = p_left->iter_idx(liter.get_index());
+              p_left->journal_inner_update(mut_liter, replacement_l->get_laddr(),
+                                         p_left->maybe_get_delta_buffer());
+              p_left->journal_inner_replace(mut_liter + 1, replacement_r->get_laddr(),
+                           replacement_pivot, p_left->maybe_get_delta_buffer());
+            } else if (riter.get_index() == p_left->get_node_size()) {
+              auto mut_liter = p_left->iter_idx(liter.get_index());
+              p_left->journal_inner_update(mut_liter, replacement_l->get_laddr(),
+                                         p_left->maybe_get_delta_buffer());
+              auto  mut_riter = p_right->iter_idx(0);
+              p_right->journal_inner_replace(mut_riter, replacement_r->get_laddr(),
+                                     replacement_pivot, p_right->maybe_get_delta_buffer());
+              p_pivot == replacement_pivot;
+            } else {  //right
+              auto mut_liter = p_right->iter_idx(liter.get_index() - p_left->get_node_size());
+              p_right->journal_inner_update(mut_liter, replacement_l->get_laddr(),
+                                           p_right->maybe_get_delta_buffer());
+              p_right->journal_inner_replace(mut_liter + 1, replacement_r->get_laddr(),
+                                    replacement_pivot, p_right->maybe_get_delta_buffer());
+            }
+            std::list<laddr_t> dec_laddrs;
```

Comment:
I think you can do std::list<laddr_t> dec_laddrs{l->get_laddr(), r->get_laddr()}